util/patch/utils.go

```diff
@@ -45,3 +46,17 @@ func unstructuredHasStatus(u *unstructured.Unstructured) bool {
 	_, ok := u.Object["status"]
 	return ok
 }
+
+func toUnstructured(obj runtime.Object) (*unstructured.Unstructured, error) {
+	// If the incoming object is already unstructured, perform a deep copy first
+	// otherwise DefaultUnstructuredConverter ends up returning the inner map without
+	// making a copy.
+	if _, ok := obj.(runtime.Unstructured); ok {
+		obj = obj.DeepCopyObject()
```

Comment:
Can we just deep copy & return the copy here?

Reply:
The runtime.Unstructured is an interface in this case, and it doesn't have a `DeepCopy()` method that returns `unstructured.Unstructured()`

Reply:
I think I have a way to make this work (using runtime.Unstructured in most places in the patch helper). Interested?